path_login.go

```
@@ -116,18 +116,31 @@ func (b *backend) pathLoginUpdate(ctx context.Context, req *logical.Request, d *
 	inner := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		raw := r.Context().Value(spnego.CTXKeyCredentials)
 		if raw == nil {
-			b.Logger().Debug("identity is nil")
-			w.Write([]byte("identity credentials are not included"))
+			_, _ = w.Write([]byte("identity credentials are not included"))
+			w.WriteHeader(400)
```

Ignoring the error seemed "wrong" to me, but alas after quick search in our code I wasn't able to find examples otherwise 🤔 

My search did lead me to this though:

>   // Write writes the data to the connection as part of an HTTP reply.
    //
    // If WriteHeader has not yet been called, Write calls
    // WriteHeader(http.StatusOK) before writing the data.

- https://golang.org/pkg/net/http/#ResponseWriter

Because we are writing a `400` status, if we move `w.WriteHeader` above the `w.Write` line, we'll reduce the number of calls to `WriteHeader` by 50%! Today I learned 😄

---

Totally agree, I hate ignoring errors. In this case, I figure that if for some reason we don't succeed writing a message to the response body, I still at least want to return a 400. Not much else I can do to communicate back to the user. 

Interesting! I knew that at the end of the day if I had never called WriteHeader it would default to a 200 response, but I _didn't_ know it was doing that as part of the Write call, that's good to know! I'll update that.